tmail-backend/event-bus-redis/src/main/java/org/apache/james/events/RedisKeyRegistrationHandler.java

```diff
@@ -181,15 +183,30 @@ private boolean isLocalSynchronousListeners(EventBusId eventBusId, EventListener
             listener.getExecutionMode().equals(EventListener.ExecutionMode.SYNCHRONOUS);
     }
 
-    private Event toEvent(String eventAsJson) {
-        return eventSerializer.asEvent(eventAsJson);
+    private List<Event> toEvent(String eventAsJson) {
+        // if the json is an array, we have multiple events
+        if (StringUtils.trim(eventAsJson).startsWith("[")) {
```

Comment:
If `eventAsJson` is in "json" format, checking if it starts with `[` is easiest. However, this approach fails if `eventAsJson` is not in JSON format, such as when `TestEventSerializer` uses `Collectors.joining("^")`. Therefore, I suggest implementing an additional `try-catch` block.